operator/internal/central/extensions/reconcile_pvc.go

```diff
@@ -200,7 +264,22 @@ func (r *reconcilePVCExtensionRun) handleDelete() error {
 }
 
 func (r *reconcilePVCExtensionRun) handleCreate(claimName string, pvcConfig *platform.PersistentVolumeClaim) error {
-	size, err := parseResourceQuantityOr(pvcConfig.Size, defaultPVCSize)
+	// Before creating a PVC, verify if prerequisites are met. Currently there
+	// is only one requirement, a default storage class must exists or a
+	// storage class has to be specified explicitly. Since it's highly specific
+	// for PVCs only, it's implemented inside the extension, instead of
+	// collecting this information at the start and passing it into the
+	// extension.
+	//
+	// Note that to make this check less disruptive, in case if we face an
+	// error we still try to create a PVC.
+	hasDefault, err := utils.HasDefaultStorageClass(r.ctx, r.client)
+	if err == nil && !hasDefault && pvcConfig.StorageClassName == nil {
+		r.log.Info("No default storage class or explicit storage class found, skip PVC creation")
+		return nil
+	}
+
+	size, err := parseResourceQuantityOr(pvcConfig.Size, r.defaults.Size)
```

Comment:
Please constrain this part to the backup PVC. There might be cases out there where the main PVC exists despite lack of storage class, and this logic will cause the operator to immediately remove such PVC (`persistence==nil` case), this is way to dangerous.

Reply:
Hm, ok, but I think such situations (no default storage class, while having a volume) is worth at least logging something verbose.